### PR TITLE
Bruce/passport rebase6

### PR DESCRIPTION
**Pull request by bvillaroman**

### CODE REVIEW

**Comment by justintmckibben** [comment]
Mind if you dropped a PR description of what all is happening here? and squashing the commits into one?

File: src/pages/PassportRedemption/style.tsx [hunk 1]
```diff
@@ -94,6 +97,70 @@ const ErrorMessage = styled.div`
...
...7786 4.91117 4.89785 7.19117 4.89785 10.0178C4.89785 12.8445 7.17786 15.1245 10.0045 15.1245C12.8312 15.1245 15.1112 12.8445 15.1112 10.0178C15.1112 7.19117 12.8312 4.91117 10.0045 4.91117ZM10.0043 13.338C8.17763 13.338 6.68429 11.8491 6.68429 10.018C6.68429 8.18686 8.17319 6.69797 10.0043 6.69797C11.8354 6.69797 13.3243 8.18686 13.3243 10.018C13.3243 11.8491 11.831 13.338 10.0043 13.338ZM16.5112 4.7022C16.5112 5.36442 15.9778 5.89331 15.3201 5.89331C14.6578 5.89331 14.129 5.35998 14.129 4.7022C14.129 4.04442 14.6623 3.51109 15.3201 3.51109C15.9778 3.51109 16.5112 4.04442 16.5112 4.7022ZM19.8937 5.9114C19.8182 4.31584 19.4537 2.90251 18.2848 1.73806C17.1204 0.573618 15.7071 0.209174 14.1115 0.129174C12.4671 0.0358402 7.53817 0.0358402 5.89373 0.129174C4.30262 0.204729 2.88928 0.569174 1.72039 1.73362C0.551504 2.89806 0.191504 4.3114 0.111504 5.90695C0.0181706 7.5514 0.0181706 12.4803 0.111504 14.1247C0.187059 15.7203 0.551504 17.1336 1.72039 18.2981C2.88928 19.4625 4.29817 19.827 5.89373 19.907C7.53817 20.0003 12.4671 20.0003 14.1115 19.907C15.7071 19.8314 17.1204 19.467 18.2848 18.2981C19.4493 17.1336 19.8137 15.7203 19.8937 14.1247C19.9871 12.4803 19.9871 7.55584 19.8937 5.9114Z"
```
Comment (justintmckibben) [inline src/pages/PassportRedemption/style.tsx:150]: nit: is it possible to pull this out into a separate file?

File: src/pages/PassportRedemption/style.tsx [hunk 2]
```diff
@@ -94,6 +97,70 @@ const ErrorMessage = styled.div`
...
```
Comment (justintmckibben) [inline src/pages/PassportRedemption/style.tsx:120]: same

File: src/pages/PassportRedemption/TicketRow.tsx [hunk 3]
```diff
@@ -51,15 +51,30 @@ const TicketRow = (props: Props) => {
       case RowStatuses.Active:
         return `READY TO REDEEM`;
       default:
-        return `${3 - props.stamps.length} MORE STAMPS UNTIL YOUR NEXT REWARD`;
+        if (props.stamps.length === 0) return;
+        return `${3 - props.stamps.length} MORE ${
+          3 - props.stamps.length > 1 ? 'STAMPS' : 'STAMP'
```
Comment (justintmckibben) [inline src/pages/PassportRedemption/TicketRow.tsx:56]: I wonder if we could add a util that pluralizes words — I've seen this ternary littered about our code

File: src/pages/PassportRedemption/RedemptionClaimScreen.tsx [hunk 4]
```diff
@@ -132,7 +135,10 @@ const PassportRedemptionClaim = ({ setCurrentScreenView }: Props) => {
           <ContentContainer>
             <CardText size="15px" color="#a8192e">
               Offer must be used in:
-              <span style={{fontWeight:'bold'}}> {formatTime(timeLeft)}</span>
+              <span style={{ fontWeight: 'bold' }}>
```
Comment (justintmckibben) [inline src/pages/PassportRedemption/RedemptionClaimScreen.tsx:138]: nit: no inline styles

File: src/pages/PassportRedemption/Passport.tsx [hunk 5]
```diff
@@ -69,7 +69,14 @@ const Passport = (props: Props) => {
...
           if (passportTickets.length > 0) {
-            setTickets(passportTickets);
+            const sortedTickets = passportTickets.sort((a, b) => {
+              const dateA = new Date(a.associated_with_contact_at);
+              const dateB = new Date(b.associated_with_contact_at);
+
+              return dateB.getTime() - dateA.getTime();
+            });
+
+            setTickets(sortedTickets.reverse());
```
Comment (justintmckibben) [inline src/pages/PassportRedemption/Passport.tsx:79]: How come we're sorting then reversing? Possible to just reverse the sort? Also I think sorts like this should be handled in the backend

File: src/pages/PassportRedemption/TicketRow.tsx (hunk 3 above)
Comment (bvillaroman) [reply src/pages/PassportRedemption/TicketRow.tsx:56]: i could do this after we release passport v1

File: src/pages/PassportRedemption/RedemptionClaimScreen.tsx (hunk 4 above)
Comment (bvillaroman) [reply src/pages/PassportRedemption/RedemptionClaimScreen.tsx:138]: @oliviasztanga

File: src/pages/PassportRedemption/Passport.tsx (hunk 5 above)
Comment (bvillaroman) [reply src/pages/PassportRedemption/Passport.tsx:79]: for some reason, the sorting i do above only returns from latest to earliest, but then if i do the reverse in the sorting algorithm, it wouldnt give me the correct order. So i just did it like this

**Pull request closed without merging**